docs/best-practices/index.md

```diff
@@ -0,0 +1,438 @@
+```terraform
+variable "populations" {
+  type = list(object({
+    name        = string
+    description = optional(string)
+  }))
+
+  default = [
+    {
+      name        = "My awesome population"
+      description = "My awesome population for awesome people"
+    },
+    {
+      name = "My awesome second population"
+    }
+  ]
+}
+
+resource "pingone_population" "my_awesome_population_list_of_objects" {
+  environment_id = pingone_environment.my_environment.id
+
+  for_each = { for population in var.populations : population.name => population }
+
+  name        = each.key
+  description = each.value.description
+}
+```
+
+The above results in creation of two unique resources:
+
+* `pingone_population.my_awesome_population_list_of_objects["My awesome population"]`
+* `pingone_population.my_awesome_population_list_of_objects["My awesome second population"]`
+
+Notice that, if the name of `My awesome population` is changed to `My awesome first population` in the variable, Terraform wants to destroy that population and re-create it with it's new index value.  While this is an unnecessary way to change the population name, destruction of populations may put user data at risk.
+
+It would be better practice therefore to use a map of objects, where there is a static key:
+```terraform
+variable "populations" {
+  type = map(object({
+    name        = string
+    description = optional(string)
+  }))
+
+  default = {
+    "first_population" = {
+      name        = "My awesome population"
+      description = "My awesome population for awesome people"
+    },
+    "second_population" = {
+      name = "My awesome second population"
+    }
+  }
+}
+
+resource "pingone_population" "my_awesome_population_map_of_objects" {
+  environment_id = pingone_environment.my_environment.id
+
+  for_each = var.populations
+
+  name        = each.value.name
+  description = each.value.description
+}
+```
+
+The above results in creation of two unique resources:
+
+* `pingone_population.my_awesome_population_list_of_objects["first_population"]`
+* `pingone_population.my_awesome_population_list_of_objects["second_population"]`
+
+In this case, if the `name` or `description` of any population changes, Terraform will correctly update the impacted resource, rather than potentially forcing a re-creation.
+
+Notice also, that if the order of the key/object pairs changes in the map, Terraform correctly calculates that there are no changes to the data with the objects themselves, because the relation of object to map key hasn't changed.  This is advantage of using `for_each` over `count`, where changing the order of items does impact the plan that Terraform calculates, because the counted index related to the data has changed.
```

Comment:
key has not changed. **This consistency is** the advantage of ...  items **impacts the plan that is calculated** because...